rsconf/component/sirepo_jupyterhub.py

```diff
@@ -0,0 +1,205 @@
+from pykern.pkdebug import pkdp
+from rsconf import component
+
+_COOKIE_SECRET = 'sirepo_jupyterhub_cookie_secret'
+_CONF_F = 'conf.py'
+_DEFAULT_PORT_BASE = 8100
+# POSIT: rsdockerspawner._DEFAULT_POOL_NAME
+_DEFAULT_USER_GROUP = 'everybody'
+_DEFAULT_POOL_V3 = _DEFAULT_USER_GROUP
+_DOCKER_TLS_SUBDIR = 'docker_tls'
+_PROXY_AUTH = 'sirepo_jupyterhub_proxy_auth'
+_TEMPLATE_D = 'template'
+
+
+class T(component.T):
+
+    def internal_build_compile(self):
+        from rsconf import db
+        from rsconf import systemd
+        from rsconf.component import docker_registry
+
+        self.buildt.require_component('docker')
+        jc, z = self.j2_ctx_init()
+        self.__run_d = systemd.docker_unit_prepare(self, jc)
+
+    def internal_build_write(self):
+        from rsconf import db
+        from rsconf import systemd
+        from rsconf.component import docker
+        from rsconf.component import docker_registry
+
+        def _env_ok(elem):
+            for p in (
+                    # TODO(e-carlin): all of auth and smtp is really too much.
+                    # We just need auth_methods for checking if the cookies is
+                    # valid. But bringing in methods then initializes the methods
+                    # so we have to bring in their full config.
+                    'SIREPO_AUTH',
+                    'SIREPO_COOKIE',
+                    'SIREPO_FEATURE_PROPRIETARY_SIM',
+                    'SIREPO_SIM_API_JUPYTERHUBLOGIN',
+                    'SIREPO_SMTP',
+                    'SIREPO_SRDB',
+            ):
+                if elem[0].startswith(f'{p}'):
+                    return True
+            return False
+
+        jc = self.j2_ctx
+        z = jc[self.name]
+        s = jc.sirepo
+        z.update(
+            user_d=s.sim_api.jupyterhublogin.user_db_root_d,
+            jupyter_docker_image=docker_registry.absolute_image(
+                jc, z.jupyter_docker_image,
+            ),
+            run_u=jc.rsconf_db.run_u,
+            jupyter_run_u=jc.rsconf_db.run_u,
+        )
+        z.setdefault('http_timeout', 30);
+        z.setdefault('template_vars', {});
+        z.template_d = self.__run_d.join(_TEMPLATE_D)
+        z.home_d = db.user_home_path(jc, z.jupyter_run_u)
+        z.cookie_secret_hex = self.secret_path_value(
+            _COOKIE_SECRET,
+            gen_secret=lambda: db.random_string(length=64, is_hex=True),
+            visibility='channel',
+        )[0]
+        z.proxy_auth_token = self.secret_path_value(
+            _PROXY_AUTH,
+            gen_secret=lambda: db.random_string(length=64, is_hex=True),
+            visibility='channel',
+        )[0]
+        z.admin_users_str = _list_to_str(z.admin_users)
+        self._rsdockerspawner(jc, z)
+
+        self.install_access(mode='711', owner=z.run_u)
+        self.install_directory(self.__run_d)
+        self.install_access(mode='700', owner=z.jupyter_run_u)
+        if z.user_d.dirname == '/srv/jupyterhub':
```

Comment:
This is annoying. I don't have a solution for it, and we'll need it for non-sirepo jupyter servers until we solve the problem of choosing pools.